### ISSUE TITLE
Install more securely using git cloning, rather than piping executable code from curl to /bin/sh

### ISSUE THREAD

**Issue by alphapapa** [meteor/meteor#2248]
In this "day and age", you're telling people to pipe curl directly to sh?  You're here on GitHub with a git repo which provides commit hashing and even cryptographically signed git tags, and you're telling people to pipe a URL directly to `/bin/sh`?  _facepalm_

**Comment by alphapapa** [comment]
It's been a while since you posted that comment.  But as far as this goes:

> But that's a bit much to ask of the random OS X user that just wants to come to meteor.com and install the tools.

I think that's a poor attitude for many reasons.

For one thing, I don't use OS X.  I run Linux.  I install official Debian packages as much as possible.  Those are not secured with TLS and CAs--those are secured with GPG.  Are you only interested in Mac users?  Web servers certainly don't run OS X.

When that's not possible, I build my own packages.  Even if it's just with `checkinstall`, that at least integrates it into the system package manager and lets me uninstall it cleanly, as well as install it on other systems simply.  How would I cleanly uninstall after "curl | sh"?  I prefer not to litter my filesystem with entries I'd have to go delete manually.  That's one of the reasons I stopped using Windows many years ago.

As far as git cloning goes, even if the initial clone isn't cryptographically secure, subsequent pulls are much safer than downloading yet another tarball, or piping yet another script through `sh`.

Frankly, I don't think you should be touting TLS and CAs.  Those do not have stellar security or trust records.

And besides that, TLS only protects against MITM attacks.  The most serious risk comes from a compromise of GitHub or GitHub user credentials.  TLS and CAs do nothing to protect against that.

You should be building binary packages and signing them with your own keys.  For source, you should be signing git tags with your own keys.

The bottom line is, all that talk about TLS and CAs seems like handwaving.  It shows a disregard for basic security, and perhaps a lack of understanding.  It certainly suggests a missing of the point.  And when you're talking about web frameworks, security is extremely important.  If the developers don't understand the difference between downloading over TLS and distributing signed packages, they probably don't understand security well enough to secure their own software.  At least, that's the assumption one should make.

So if I were looking for a framework like yours, I'd run for the hills upon seeing "install our software with curl | sh".  But, hey, it's your project... :)
